ide/editor.lib2/src/org/netbeans/modules/editor/lib2/view/HighlightsViewUtils.java

```diff
@@ -777,4 +756,85 @@ static View breakView(int axis, int breakPartStartOffset, float x, float len,
         return null;
     }
 
+    // Package-private for testing.
+    /**
+     * Calculate the position at which to break a line in a paragraph. A break offset of X means
+     * that the character with index (X-1) in {@code paragraph} will be the last one on the physical
+     * line.
+     *
+     * <p>The current implementation avoids creating lines with leading whitespace (when words are
+     * separated by at most one whitespace character), allows lines to be broken after hyphens, and,
+     * if {@code allowWhitespaceBeyondEnd} is true, allows one whitespace character to extend beyond
+     * the preferred break width to make use of all available horizontal space. Very long
+     * unbreakable words may extend beyond the preferred break offset regardless of the setting of
+     * {@code allowWhitespaceBeyondEnd}.
+     *
+     * <p>It was previously considered to allow an arbitrary number of whitespace characters to
+     * trail off the end of each wrap line, rather than just one. In the end, it turned out to be
+     * better to limit this to just one character, as this conveniently avoids the need to ever
+     * position the visual text caret outside the word-wrapped editor viewport (except in cases of
+     * very long unbreakable words).
+     *
+     * @param paragraph a long line of text to be broken, i.e. a paragraph, or the remainder of a
+     *        paragraph if some of its initial lines of wrapped text have already been laid out
+     * @param preferredMaximumBreakOffset the preferred maximum break offset
+     * @param allowWhitespaceBeyondEnd if true, allow one whitespace character to extend beyond
+     *        {@code preferredMaximumBreakOffset} even when this could be avoided by choosing a
+     *        smaller break offset
+     */
+    static int adjustBreakOffsetToWord(CharSequence paragraph,
+            final int preferredMaximumBreakOffset, boolean allowWhitespaceBeyondEnd)
+    {
+        if (preferredMaximumBreakOffset < 0)
+            throw new IllegalArgumentException();
+        if (preferredMaximumBreakOffset > paragraph.length())
+            throw new IllegalArgumentException();
+        /* BreakIterator.getLineInstance already seems to have a cache; creating a new instance here
+        is just the cost of BreakIterator.clone(). So don't bother trying to cache the BreakIterator
+        here. */
+        BreakIterator bi = BreakIterator.getLineInstance(Locale.US);
```

Comment:
Really explicitly `Locale.US` ?

Reply:
It should be sensible for source code and space-separated plain text in non-Asian languages. Languages like Thai, which don't use spaces, would need a locale-specific dictionary-based BreakIterator.

An alternative would be to use the system locale, but I'm not sure it's more appropriate. A Thai user is still likely to use NetBeans for programming languages that break better using the US locale.

Yet another alternative would be to add a client property on the EditorPane to set the word wrapping locale. That would be useful for platform applications that wants to use the NetBeans EditorKit for editing of natural language text.

Reply:
OK. Maybe just file a RFE for the localized wordwrap; I am sure that there are more places, which should be changed appropriately to get word-aware features fully localized.